assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py

```diff
@@ -40,17 +64,31 @@ def evaluate_metrics_threshold(
         is_not_nan_metrics_threshold_df.metric_value.isNotNull()
     )
 
-    metrics_threshold_breached_df = is_not_nan_metrics_threshold_df.where(
-        F.col("metric_value") > F.col("threshold_value")
-    )
-
+    metrics_threshold_breached_df = calculate_metrics_breach(is_not_nan_metrics_threshold_df)
     metrics_threshold_breached_df.show()
 
+    return send_email_for_breached(metrics_threshold_breached_df, notification_emails, signal_name)
+
+
+def calculate_metrics_breach(metrics_threshold_df: pyspark.sql.DataFrame):
+    """Calculate the breached metrics by the given thresholds."""
+    metrics_threshold_breached_df = metrics_threshold_df.where(
+        (F.col(SIGNAL_METRICS_METRIC_NAME).isin(Metric_Value_Should_Greater_Than_Threshold) &
+         (F.col(SIGNAL_METRICS_METRIC_VALUE) < F.col(SIGNAL_METRICS_THRESHOLD_VALUE))) |
+        (~F.col(SIGNAL_METRICS_METRIC_NAME).isin(Metric_Value_Should_Greater_Than_Threshold) &
+         (F.col(SIGNAL_METRICS_METRIC_VALUE) > F.col(SIGNAL_METRICS_THRESHOLD_VALUE)))
```

Comment:
how about == ?

Reply:
We do not need to consider the "equal" when evaluating the value of the metric towards the threshold. If the value equals the threshold, we consider it not breaching. This is also consistent with the UX criteria.